styx-common/src/main/java/com/spotify/styx/state/StateData.java

```diff
@@ -46,6 +47,7 @@
   Optional<String> triggerId(); //for backwards compatibility
   Optional<String> executionId();
   Optional<ExecutionDescription> executionDescription();
+  Optional<Set<String>> resourceIds(); // resources referenced in the workflow configuration at the time of dequeue
```

Comment:
Don't we need to persist this in `DatastoreStorage`?

Reply:
Yes, because we need it for two actions `increment counters` for the given resources whenever a wfi is dequeued, and `decrement counters` whenever it is terminated or failed. Those actions may be done by two different scheduler instances, and we need the resourceIds for both.

Reply:
Yes, addressed in additional commit